R/sample.R

```diff
@@ -97,3 +103,100 @@ mcstate_sampler <- function(name, initialise, step, finalise) {
   class(ret) <- "mcstate_sampler"
   ret
 }
+
+
+initial_parameters <- function(initial, model, rng, call = NULL) {
+  n_pars <- length(model$parameters)
+  n_chains <- length(rng)
+  if (is.null(initial)) {
+    ## Really this would just be from the prior; we can't directly
+    ## sample from the posterior!
+    initial <- lapply(rng, function(r) model$direct_sample(r))
```

Comment:
i am assuming this is not guaranteed to be able to take in a vector?

Reply:
I don't understand the questions here (or which bit of this is not taking in a vector)

Reply:
ahh dw, was talking about doing `model$direct_sample(rng)` instead of the lapply but yh i dont think that will always work